test/tracing/test_set_direction.py

```diff
@@ -0,0 +1,76 @@
+#  Copyright 2021 Zeppelin Bend Pty Ltd
+#
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from zepben.evolve import TestNetworkBuilder, Feeder, ConductingEquipment, set_direction, FeederDirection, NetworkService
+
+
+@pytest.mark.asyncio
+async def test_set_direction():
+    network = _get_single_feeder_network()
+    await set_direction().run(network)
+
+    assert network.get("b0-t1").normal_feeder_direction == FeederDirection.NONE
+    assert network.get("b0-t2").normal_feeder_direction == FeederDirection.DOWNSTREAM
+    assert network.get("c1-t1").normal_feeder_direction == FeederDirection.UPSTREAM
+    assert network.get("c1-t2").normal_feeder_direction == FeederDirection.DOWNSTREAM
+    assert network.get("j2-t1").normal_feeder_direction == FeederDirection.UPSTREAM
+    assert network.get("j2-t2").normal_feeder_direction == FeederDirection.DOWNSTREAM
+
+
+@pytest.mark.asyncio
+async def test_set_direction_doesnt_flow_through_feeder_heads():
+    network = _get_dual_parallel_feeders_network()
+    await set_direction().run(network)
```

Comment:
This can be combined with the feeder assignment into a call to `TestNetworkBuilder.build`